### DISCUSSION TITLE
Deployment error on vercel

### DISCUSSION THREAD

**Discussion by mohammga** [vercel/next.js#56204] [Help] [answered]
### Summary

Deployment error on vercel and it says: Attempted import error: 'authOptions' is not exported from './auth/[...nextauth]/route' (imported as 'authOptions').

### Additional information

```js
-----------------------------------------------------------------------------------------------
lib/authOptions.js
-----------------------------------------------------------------------------------------------

import { prisma } from '@/lib/prisma';
import { compare } from 'bcrypt';
import CredentialsProvider from 'next-auth/providers/credentials';

export const authOptions = {
 ....
};

-----------------------------------------------------------------------------------------------
auth/[...nextauth]/route
-----------------------------------------------------------------------------------------------

import { authOptions } from '@/lib/authOptions'
import NextAuth from 'next-auth';

const handler = NextAuth(authOptions);
export { handler as GET, handler as POST };
```

**Comment by icyJoseph** [comment]
Hi,

Maybe it is an issue with your example, but you have `lib/authOption`, without ending `s`, but you import:

```js
import { authOptions } from '@/lib/authOptions' // <--- this one ends with Options not Option
```

With a trailing `s`.

**Reply by mohammga** [reply]
I wrote it wrong in the question.
it should be lib/authOptions.js

import { authOptions } from '@/lib/authOptions'

It works on localhost and on vercel but on deployment to vercel the build gives me this error: "Deployment error on vercel and it says: Attempted import error: 'authOptions' is not exported from './auth/[...nextauth]/route' (imported as 'authOptions')."

"I have also tried to change the name of the file to auth", same problem :(

Logs from the build:
[16:00:54.590] ./app/api/route.js
[16:00:54.590] Attempted import error: 'authOptions' is not exported from './auth/[...nextauth]/route' (imported as 'authOptions').
[16:00:54.590]
[16:00:54.590] Import trace for requested module:
[16:00:54.590] ./app/api/route.js

**Reply by icyJoseph** [reply, marked as answer]
But wait, you are importing to `app/api/route.js`, from `./auth/[...nextauth]/route`?

Well then it makes sense, because as you have shown:

```ts
export { handler as GET, handler as POST };
```

That's all you export from `./auth/[...nextauth]/route`.  

Could we see `./app/api/route.js`?

**Reply by mohammga** [reply]
Found the error: 
```js
import { getServerSession } from 'next-auth/next';
import { NextResponse } from 'next/server';
import { authOptions } from './auth/[...nextauth]/route';

export async function GET(req) {
  const session = await getServerSession(authOptions);

  if (!session) {
    return new NextResponse(JSON.stringify({ error: 'unauthorized' }), {
      status: 401,
    });
  }

  console.log('GET API', session);
  return NextResponse.json({ authenticated: !!session });
}
```
 This: 
```js
import { authOptions } from './auth/[...nextauth]/route';
```
Should be:
```js
import { authOptions } from '@/lib/authOptions'
```
